packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerMigrationUtils.ts

```diff
@@ -0,0 +1,191 @@
+import { $TSContext, exitOnNextTick, JSONUtilities, pathManager, stateManager } from 'amplify-cli-core';
+import chalk from 'chalk';
+import * as fs from 'fs-extra';
+import _ from 'lodash';
+import * as path from 'path';
+import { categoryName } from '../../../constants';
+import { layerConfigurationFileName, LegacyFilename, versionHash } from './constants';
+import { loadPluginFromFactory } from './functionPluginLoader';
+import { writeLayerConfigurationFile } from './layerConfiguration';
+import { defaultLayerPermission, LayerPermission, LayerRuntime, PermissionEnum } from './layerParams';
+
+export const enum LegacyPermissionEnum {
```

Comment:
This would require wide sweeping changes that affect dozens of files to remain consistent. Going to leave as is since we use "legacy" throughout the code base.